python/cuml/dask/common/input_utils.py

```diff
@@ -0,0 +1,314 @@
+#
+# Copyright (c) 2020, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+import cudf
+import cupy as cp
+import dask.array as da
+
+from cuml.dask.common.dask_df_utils import to_dask_cudf
+from cuml.utils.memory_utils import with_cupy_rmm
+
+from collections import OrderedDict
+from cudf.core import DataFrame
+from dask_cudf.core import DataFrame as dcDataFrame
+
+from dask import delayed
+from dask.distributed import wait
+from dask.distributed import default_client
+from tornado import gen
+from toolz import first
+from collections.abc import Iterable
+from dask.array.core import Array as daskArray
+from uuid import uuid1
+
+from collections import defaultdict
+from functools import reduce
+
+import dask.dataframe as dd
+
+
+class MGData:
```

Comment:
Should it be MGData or MNMGData? or DistributedData?

Reply:
Initially I had it as DistributedData, but that is such a verbose and ugly long word with 2 consecutive `d`s, so I didn't like it. MNMGData sounds good to me, of course any other suggestions accepted